ext/standard/math.c

```diff
@@ -228,6 +228,48 @@ PHP_FUNCTION(abs)
 }
 /* }}} */
 
+/* {{{ Return the given value if in range of min and max */
+PHP_FUNCTION(clamp)
+{
+	zval *zvalue, *zmin, *zmax;
+
+	ZEND_PARSE_PARAMETERS_START(3, 3)
+		Z_PARAM_NUMBER(zvalue)
+		Z_PARAM_NUMBER(zmin)
+		Z_PARAM_NUMBER(zmax)
+	ZEND_PARSE_PARAMETERS_END();
+
+	if (Z_TYPE_P(zmin) == IS_DOUBLE) {
+		if (UNEXPECTED(zend_isnan(Z_DVAL_P(zmin)))) {
+			zend_argument_type_error(2, "cannot be of type NAN");
+			RETURN_THROWS();
+		}
+	} else if (Z_TYPE_P(zmax) == IS_DOUBLE) {
+		if (UNEXPECTED(zend_isnan(Z_DVAL_P(zmax)))) {
+			zend_argument_type_error(3, "cannot be of type NAN");
+			RETURN_THROWS();
+		}
+	} else if (Z_TYPE_P(zvalue) == IS_DOUBLE) {
+		if (UNEXPECTED(zend_isnan(Z_DVAL_P(zvalue)))) {
+			RETURN_COPY(zvalue);
+		}
+	}
+
+	if (zend_compare(zmax, zmin) == -1) {
+		zend_argument_value_error(2, "cannot be greater than Argument #3 ($max)");
```

Comment:
And then:
```suggestion
		zend_argument_value_error(2, "must be smaller than or equal to argument #3 ($max)");
```

The subtlety here is that while `$x > NAN` is false, `!($x < NAN)` is true.